types/applications/sidebarTabs/sidebarDirectory.d.ts

```diff
@@ -0,0 +1,262 @@
+/**
+ * A shared pattern for the sidebar directory which Actors, Items, and Scenes all use
+ */
+declare abstract class SidebarDirectory extends SidebarTab {
+  constructor(options?: Partial<Application.Options>);
+
+  /**
+   * References to the set of Entities which are displayed in the Sidebar
+   */
+  entities: Entity[];
+
+  /**
+   * Reference the set of Folders which exist in this Sidebar
+   * @defaultValue `null`
+   */
+  folders: Folder[];
+
+  /**
+   * @override
+   */
+  static get defaultOptions(): SidebarDirectory.Options;
+
+  /**
+   * The named entity which this Sidebar Directory contains
+   * @remarks This method is abstract in SidebarTab.
+   */
+  static get entity(): string;
+
+  /**
+   * The Entity collection which this Sidebar Directory contains
+   * @remarks This method is abstract in SidebarTab.
+   */
+  static get collection(): EntityCollection;
+
+  /**
+   * A reference to the Entity class which is displayed within this EntityCollection
+   */
+  static get cls(): ConstructorOf<Entity>;
+
+  /**
+   * Initialize the content of the directory by categorizing folders and entities into a hierarchical tree structure.
+   */
+  initialize(): void;
+
+  tree: SidebarDirectory.Tree;
+
+  /**
+   * Given an entity type and a list of entities, set up the folder tree for that entity
+   * @param folders  - The Array of Folder objects to organize
+   * @param entities - The Array of Entity objects to organize
+   * @returns A tree structure containing the folders and entities
+   */
+  static setupFolders(folders: Folder[], entities: Entity[]): SidebarDirectory.Tree;
+
+  /**
+   * Populate a single folder with child folders and content
+   * This method is called recursively when building the folder tree
+   */
+  protected static _populate(
+    folder: Folder,
+    folders: Folder[],
+    entities: Entity[],
+    {
+      allowChildren
+    }: {
+      /**
+       * @defaultValue `true`
```

Comment:
I don't think this is shown in the documentation. Maybe create an `@param` in the method's TSDoc?

Reply:
It does show in the documentation, but only when you are already in the completion/hover of the object. At the same time the doc for the method doesn't show anymore in that place. But it should be consistent, so I moved it to the method.